module/zstd/zstd.c

```diff
@@ -295,24 +295,43 @@ zstd_enum_to_cookie(enum zio_zstd_levels level, int16_t *cookie)
 	/* Invalid/unknown ZSTD level - this should never happen. */
 	return (1);
 }
+typedef struct {
```

Comment:
hmm, that could be simplyfied:

~~~c
/* NOTE: all fields in this header are in big endian order */
struct zstd_header {
  /* contains compressed size */
  uint32_t size;

  /*
   * contains the version and level
   * we have to choose a union here to handle
   * endian conversation since the version and level
   * is bitmask encoded.
   */
  union {
    uint32_t raw_version_level;
    struct {
      uint32_t version : 24;
      uint8_t level;
    };
  };

  char data[];
};
~~~

Reply:
yes. this works too. i just made a separate struct out of it for readability. and a earlier version used that struct direct

Reply:
IMHO readability is way better this way :) but that's personal preference I guess